src/seedu/addressbook/AddressBook.java

```diff
@@ -29,1158 +30,1260 @@
+	 * if the parameter name is self-explanatory.
+	 * In the method below, '@param userInput' comment has been omitted.
+	 * ====================================================================
+	 */
+	/**
+	 * Echoes the user input back to the user.
+	 */
+	private static void echoUserCommand(String userCommand)
+	{
+		showToUser("[Command entered:" + userCommand + "]");
+	}
+
+	/*
+	 * ==============NOTE TO STUDENTS==========================================
+	 * If the reader wants a deeper understanding of the solution, she can go
+	 * to the next level of abstraction by reading the methods (given below)
+	 * that is referenced by the method above.
+	 * ====================================================================
+	 */
+
+	/**
+	 * Processes the program main method run arguments.
+	 * If a valid storage file is specified, sets up that file for storage.
+	 * Otherwise sets up the default file for storage.
+	 *
+	 * @param args full program arguments passed to application main method
+	 */
+	private static void processProgramArgs(String[] args)
+	{
+		if(args.length >= 2)
+		{
+			showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+			exitProgram();
+		}
+
+		if(args.length == 1)
+		{
+			setupGivenFileForStorage(args[0]);
+		}
+
+		if(args.length == 0)
+		{
+			setupDefaultFileForStorage();
+		}
+	}
+
+	/**
+	 * Sets up the storage file based on the supplied file path.
+	 * Creates the file if it is missing.
+	 * Exits if the file name is not acceptable.
+	 */
+	private static void setupGivenFileForStorage(String filePath)
+	{
+		if(!isValidFilePath(filePath))
+		{
+			showToUser(String.format(MESSAGE_INVALID_FILE, filePath));
+			exitProgram();
+		}
+
+		storageFilePath = filePath;
+		createFileIfMissing(filePath);
+	}
+
+	/**
+	 * Displays the goodbye message and exits the runtime.
+	 */
+	private static void exitProgram()
+	{
+		showToUser(MESSAGE_GOODBYE, DIVIDER, DIVIDER);
+		System.exit(0);
+	}
+
+	/**
+	 * Sets up the storage based on the default file.
+	 * Creates file if missing.
+	 * Exits program if the file cannot be created.
+	 */
+	private static void setupDefaultFileForStorage()
+	{
+		showToUser(MESSAGE_USING_DEFAULT_FILE);
+		storageFilePath = DEFAULT_STORAGE_FILEPATH;
+		createFileIfMissing(storageFilePath);
+	}
+
+	/**
+	 * Returns true if the given file is acceptable.
+	 * The file path is acceptable if it ends in '.txt'
+	 * TODO: Implement a more rigorous validity checking.
+	 */
+	private static boolean isValidFilePath(String filePath)
+	{
+		return filePath.endsWith(".txt");
+	}
+
+	/**
+	 * Initialises the in-memory data using the storage file.
+	 * Assumption: The file exists.
+	 */
+	private static void loadDataFromStorage()
+	{
+		initialiseAddressBookModel(loadPersonsFromFile(storageFilePath));
+	}
+
+	/*
+	 * ===========================================
+	 *           COMMAND LOGIC
+	 * ===========================================
+	 */
+
+	/**
+	 * Executes the command as specified by the {@code userInputString}
+	 *
```

Comment:
consider enhancing the comment to say what happens if user gives an invalid input string.